internal/interface/httpapi/api.go

```diff
@@ -150,7 +159,37 @@ func (a *Api) getPage(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	http.Redirect(w, r, l, http.StatusSeeOther)
+	m := sync.Mutex{}
+	m.Lock()
+	a.RedirectRequestChannel <- RedirectRequestInput{w, r, l, &m}
+	m.Lock()
+}
+
+func CheckLink(in <-chan RedirectRequestInput) <-chan RedirectRequestInput {
+	out := make(chan RedirectRequestInput)
+	go func() {
+		for rri := range in {
+			_, err := http.Get(rri.l)
+			if err != nil {
+				rri.w.WriteHeader(http.StatusNotFound)
+				rri.m.Unlock()
```

Comment:
It is a very bad practice to lock and unlock mutex in different scopes. 
Why do you need mutex?

Reply:
It's [clarified a bit](https://github.com/mp-hl-2021/lenkeforkortelse/pull/8#discussion_r641499887) in the `neckbosov`'s comment thread. 

Should we rewrite this "hacky" solution? We didn't want it to be hacky. Probably, we have just misunderstood the idea of a "Pipeline".